src/fetch/PageAction.js

```diff
@@ -0,0 +1,113 @@
+import { pageAction } from "./prompts.js";
+import { logger } from "../log/logger.js";
+import { TagRemovingMinimizer } from "../min/TagRemovingMinimizer.js";
+import { getAI } from '../ai/index.js';
+import { Timer } from "../log/timer.js";
+
+export const PageAction = class {
+  constructor(namespace, prompt) {
+    this.namespace = namespace;
+    this.prompt = prompt;
+    this.ai = getAI()
+  }
+
+  async learn(examples) {
+
+  }
+
+  async execute(page) {
+    const timer = new Timer();
+    const url = page.url;
+    const doc = await page.content();
```

Comment:
- learn() should receive Document instances
- to get Documents, use fetcher.fetch async generator:

```
import { getFetcher } from 'src/index.js';

const fetcher = getFetcher();
const doc = await fetcher.first('https://...');
```